### PR TITLE
Phytocell

### PR DESCRIPTION
**Pull request by natascher**
documentation with files

### CODE REVIEW

File: motion/p5js/cones/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const height = 60;
+const radius = 60;
+let y = 0;
+let increment = 1;
+
+function setup() {
+  createCanvas(500, 500, WEBGL);
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: the canvas needs to be assigned to the sketch id div. See https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-P5js-Sketches#code

File: motion/p5js/cones/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  frameRate(30);
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: loose the framerate. Makes the side slow.

File: motion/p5js/cones/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  drawRotated(5);
+
+  translate(2* radius, 2 * radius);
+
+  drawRotated(3);
+
+  translate(2 * radius, 2 * radius);
+
+  drawRotated(1);
+}
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: lots of unnecessary whitespace in this file.

File: projects/analog-phytocell/index.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
+---
+authors:
+  - "Natalie Schreiber"
+layout: project
+title: "Projekt 1 | Algorithmus | Phytocell"
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: I think "Phytocell" is enough as title.

File: projects/analog-phytocell/index.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+type: project
+---
+
+![Cover](splash.png)
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: relative paths please

File: projects/analog-phytocell/index.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+# Analog Algorithm | PhytoCell
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: The title will be added by the system https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-Projects#markdown-files--syntax

File: projects/analog-phytocell/index.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+## General
+
+_Algorithms? That reminds me of confusing and therefore boring math lessons in school.
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: Is this a quote? If so mark it as one.

File: projects/analog-phytocell/index.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+I'm done with it._
+
+You're not.
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: No slang and twitter, facebook writing style please… It is not a blog. It is a documentation of your work. Try a objective perspective on the project.

File: projects/analog-phytocell/index.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+Everyone who once admired the precise structure of a snowflake appreciates designing with algorithms.
+
+They are exact iterations in structures, enhancing themselves, building on top of another, moving away from each other, intertwining and amount as a whole to something completely new and superior.
+Well, what can be more exciting for a designer than using this ordinary template, understanding, doing research into, experimenting, and adapting it - by coding?
+
+![Sketch](assets/images/nature_code.png)
+
+## Getting in touch - the analog algorithm
+
+Our experience so far and from our sciolism is that we suspect that algorithms mostly deal with coding. But...what exactly?
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: slang…

File: projects/analog-phytocell/index.md [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+Werner Stangl, an Austrian psychologist describes an algorithm as _a systematic, logic rule or approach which leads to the solution of the problem in hand._ [1]
+
+Aha.
+Well then?
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: slang…

File: projects/analog-phytocell/index.md [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+
+Therefore we approached the answer to this question playfully in the first part of the seminar. Each student should come up with an assignment and write it down on a piece of paper. If you exactly follow the instructions you would get a certain (abstract) figure or drawing as a result - a simple analog algorithm.
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: Try a objective perspective. No we, I, my…

File: projects/analog-phytocell/index.md [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+We swapped the assignments with one another and everyone had 3 minutes to fulfill the task within the scope of the predefined rules.
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: This is not a blog where you write a story about a journey.

File: projects/analog-phytocell/index.md [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+After the 3 minutes were up we swapped again and so on and so forth.
+
+## Output - specification and variance
+
+After 6 iterations we had the results: most of the pictures deviated from the initial one that you had in mind when setting the task.
+
+![Output1](assets/images/results1.png)
```
Comment (fabianmoronzirfas) [inline projects/analog-phytocell/index.md]: relative paths please  

    Output1

File: repeat/p5js/hidden-circles/index.js [hunk 14]
```diff
@@ -0,0 +1,69 @@
+var dis = 20;
+
+function setup() {
+  createCanvas(500, 500);
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.js]: the canvas needs to be assigned see https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-P5js-Sketches#code

File: repeat/p5js/hidden-circles/index.js [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+  }
+}
+
+function draw() {
+  stroke(0);
+  strokeWeight(2);
+  drawLines(100, dis);
+  
+  strokeWeight(10);
+  stroke(255, 255, 255);
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.js]: Lots of magic numbers…

File: repeat/p5js/hidden-circles/index.js [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+  noFill();
+  ellipse(250, 250, 440);
+  
+  strokeWeight(10);
+  stroke(255, 255, 255);
+  noFill();
+  ellipse(250, 250, 500);
+}
+ 
+ function mouseClicked(){
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.js]: remove the image saving.

File: repeat/p5js/hidden-circles/index.md [hunk 17]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Natalie Schreiber
+author-url: https://github.com/natascher
+title: hidden_circles
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.md]: I already reviewed that. Hidden Circles please

File: motion/p5js/cones/index.js (hunk 3 above)
Comment (frontendphil) [inline motion/p5js/cones/index.js]: Hey, I help @natascher from time to time with her assignments and I don't get some of the code comments. Why are those white-spaces unnecessary? They separate the function which positions the canvas from the function that draws on the canvas. This way parts that logically belong together (in this case always one line) is surrounded by white-space. Code optimisations such das removing white-space should be done by a built system, not the developer.

File: repeat/p5js/hidden-circles/index.js (hunk 15 above)
Comment (frontendphil) [inline repeat/p5js/hidden-circles/index.js]: Most of these magic numbers are kind of enforced by the poor design of the API of the p5 framework. Take for example the stroke function. What do the parameters do? Seems like they are the `r`, `g`, and `b` part of a color. But can you be sure, that it is not `b`, `r`, and `g`? Normally you would counter these "magic" numbers with variables. But what would that mean?

```jsx
const strokeWeightValue = 10 // in px and value so that it does not clash with the magic global function

strokeWeight(strokeWeightValue) // not more descriptive

const strokeRedPart = 255
const strokeBluePart = 255
const strokeGreenPart = 255

stroke(strokeRedPart, strokeGreenPart, strokeBluePart) // wtf
```

A better API would make these "magic" numbers more expressive by design.

```jsx
strokeWidth({ px: 10 })

stroke({ r: 255, g: 255, b: 255 })
```

This is true for basically every function that needs to be used. The way the API of p5 is designed it forces you to some extent to fall back to magic numbers, because wrapping values in constants is cumbersome and doesn't add value.

**Comment by fabianmoronzirfas** [comment]
@natscher if you don't agree with my review please discuss it with me. 

@frontendphil I really don't need to defend my teaching against you.

**Comment by frontendphil** [comment]
@fabianmoronzirfas I feel we got off on the wrong foot :) And also @natascher had nothing to do with my comments. I just had a look at the PR out of curiosity and found that some of your code comments leave some room open for discussion. That's also why I tried to elaborate on some of the comments as for instance the magic numbers. I also face situations like these in my everyday work but as stated above I have the strong feeling that those numbers are more a symptom and not the real error. I'd also wanted to raise awareness that the API of p5 could be improved with just some small changes. 

So, no, you don't have to defend your teaching because I never questioned your teaching as I don't know you at all. But sometimes you need to defend (even though I think discuss is a better word) the comments you make.

**Comment by fabianmoronzirfas** [comment]
See the fiddle we made today 25.04.2017 in this gist https://gist.github.com/fabianmoronzirfas/a8085d44a3b47438276ae81efa7e0819

File: repeat/p5js/hidden-circles/index.md [hunk 18]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Natalie Schreiber
+author-url: https://github.com/natascher
+title: Hidden Circles Pt.2
+tags: [P5.js, repeat, Law of Connected Elements]
+permalink: /repeat/hidden-circles2/
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.md]: Can you remove the 2 from the permalink?

File: repeat/p5js/hidden-circles/index.html [hunk 19]
```diff
@@ -0,0 +1,20 @@
+<!doctype html>
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.html]: We don't need this file. Please remove

File: motion/p5js/cones/index.html [hunk 20]
```diff
@@ -0,0 +1,20 @@
+<!doctype html>
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.html]: We don't need this file. Please remove.

**Comment by fabianmoronzirfas** [comment]
merged and live https://interface.fh-potsdam.de/gestalten-in-code/projects/analog-phytocell/ && https://interface.fh-potsdam.de/gestalten-in-code/motion/cones/ https://interface.fh-potsdam.de/gestalten-in-code/repeat/hidden-circles/
